src/azure/cli/commands/arm.py

```diff
@@ -394,12 +406,25 @@ def _update_instance(instance, part, path):
     try:
         index = index_regex.match(part)
```

Comment:
>index_regex [](start = 16, length = 11)

minor: it is a little confusing to have an index that contains a key value filter.  Consider renaming this